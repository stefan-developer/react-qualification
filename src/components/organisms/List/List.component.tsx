import React from "react";
import * as Styled from "./List.styles";
import IGithubRepo from "../../../models/GithubRepo";

interface ListProps {
  data: IGithubRepo[];
}

function List(props: ListProps): JSX.Element {
  const { data } = props;
  return (
    <Styled.Wrapper>
      {data.map((repo: any) => (
        <li key={repo.name}>
          <a href={repo.html_url}>{repo.name}</a>
        </li>
      ))}
    </Styled.Wrapper>
  );
}
export default List as React.FC<ListProps>;
