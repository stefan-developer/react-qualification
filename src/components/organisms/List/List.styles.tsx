import styled from "styled-components";

export const Wrapper = styled.ul`
  width: 100%;
  max-width: 600px;
  li {
    a {
      font-family: "Open Sans", sans-serif;
    }
  }
`;
