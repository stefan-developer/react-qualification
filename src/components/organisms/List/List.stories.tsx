import React from "react";

import { storiesOf } from "@storybook/react";

import List from "./List.component";
import IGithubRepo from "../../../models/GithubRepo";

const data = [
  { name: "my" },
  { name: "github" },
  { name: "repo" },
  { name: "list" },
] as IGithubRepo[];

storiesOf("List", module).add("default", () => (
  <List data={data} />
));
