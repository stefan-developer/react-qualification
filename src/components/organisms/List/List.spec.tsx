import React from "react";
import renderer from "react-test-renderer";

import IGithubRepo from "../../../models/GithubRepo";

import List from "./List.component";

test("List renders as expected", () => {
  const data = [
    { name: "my" },
    { name: "github" },
    { name: "repo" },
    { name: "list" },
  ] as IGithubRepo[];

  const component = renderer.create(<List data={data} />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
