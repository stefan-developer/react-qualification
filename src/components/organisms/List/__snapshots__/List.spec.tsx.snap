// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders as expected 1`] = `
<ul
  className="sc-AxjAm juAXlR"
>
  <li>
    <a>
      my
    </a>
  </li>
  <li>
    <a>
      github
    </a>
  </li>
  <li>
    <a>
      repo
    </a>
  </li>
  <li>
    <a>
      list
    </a>
  </li>
</ul>
`;
